Plutarch/Api/V1.hs

```diff
@@ -154,24 +183,30 @@ instance PIsDataRepr PScriptContext where
 -- General types, used by V1 and V2
 
 data PScriptPurpose (s :: S)
-  = PMinting (Term s (PDataList '[PCurrencySymbol]))
-  | PSpending (Term s (PDataList '[PTxOutRef]))
-  | PRewarding (Term s (PDataList '[PStakingCredential]))
-  | PCertifying (Term s (PDataList '[PDCert]))
+  = PMinting (Term s (PDataRecord '["_0" ':= PCurrencySymbol]))
```

Comment:
There's an opportunity to revise the naming. 
I've defaulted to numeric naming like this for everything in plutus that doesn't itself provide field names, but these could be, for example, something like 'get', or be named after their constructors.

Reply:
I think this is fine TBH. It's sort of reminiscent of Rust's "newtypes".